MetaMask/metamask-mobile
# [e2e] Send ETH to an EOA using Ganache network on Android

**Description**
This PR adds an e2e testcase for Android with wdio/appium, using the Ganache network. The test is a Simple send flow to an EOA. For that, it's done the following:

- A Confirmations folder has been created for including the android e2e tests related to Confirmations
- A Confirmations suite has been created in order to run all confirmation related tests by using the flag `--suite confirmations`
- A Ganache server will be started whenever we use `Ganache server is started` from common steps
- The Ganache server will be stopped whenever we use `Ganache server is stopped` from common steps
- Ganache sets `10000` ETH by default to the generated accounts given the seedphrase we pass it. This seedphrase is defined on ` Accounts.getValidAccount()`
  - It seems that the envar `MM_TEST_ACCOUNT_SRP` is not working and not correctly set. For this reason, running local tests we can add a seedphrase, in case the envar does not work  ` seedPhrase: process.env.MM_TEST_ACCOUNT_SRP || 'phrase upgrade clock rough situate wedding elder clever doctor stamp excess tent',`. Seems this is related to issue 762


**Manual testing steps**
- Connect an Android device to your computer / use simulator (see Android setup on readme)
  - You'll need to set the following vars
```
export ANDROID_HOME=/home/PATH_TO_ANDROID/Android/Sdk
export ANDROID_SDK_ROOT=/home/PATH_TO_ANDROID/Android/Sdk
```

- Open a terminal
- Expose the needed ports on your Android device
  - `adb reverse tcp:8081 tcp:8081`
  - `adb reverse tcp:8545 tcp:8545`  -- where ganache server runs
- Create the QA build
  - `yarn setup`
  - `yarn start:android:qa` 
- Open another terminal
  - run yarn watch 
- On the first terminal run the specific testcase `yarn test:wdio:android --spec wdio/features/Confirmations/SendEthEOA.feature`
- Notice you might need to add a seedphrase on the envar file and if it does not work, you can add it directly as indicated above
  
**Screenshots/Recordings**


![Screenshot from 2023-04-19 10-56-53](https://user-images.githubusercontent.com/54408225/233032654-76b0f6db-ad93-41a7-b3c7-cea4ee4a0570.png)

[android-ganache-send-eth.webm](https://user-images.githubusercontent.com/54408225/233032738-0592c951-b283-433c-92b3-8c2d125f38ef.webm)


**Issue**

- Resolves https://github.com/MetaMask/mobile-planning/issues/896


**Checklist**
